### PR TITLE
WebGPU: use 'auto' if pipeline layout is not specified

### LINKED ISSUE #18178: [WebGPU] Setup an auto layout in RenderPipelineDescriptor (supported?)
Hi,

When creating a new render pipeline I am not able to pass an auto layout in the render pipeline descriptor, any idea how to define it?

```c++
wgpu::RenderPipelineDescriptor desc{};
desc.layout = "auto"; // does not work, only accepts a wgpu::PipelineLayout
```
Commenting the layout line raises a warning which is not ideal: `Leaving the layout of a render pipeline undefined has been deprecated, and specifying a pipeline layout will soon be required. Set layout to 'auto' if an implicit pipeline layout is desired.`

### PR DESCRIPTION
**Pull request by austinEng**
Fixes #18178

### CODE REVIEW

**Comment by austinEng** [comment]
@kainino0x PTAL

**Review by sbc100 — APPROVED** [review]
rubberstamp lgtm

**Review by kainino0x — APPROVED** [review]
Thank you Austin and Sam!

File: src/library_webgpu.js [hunk 1]
```diff
@@ -1188,7 +1193,7 @@ var LibraryWebGPU = {
 
     var desc = {
       "label": undefined,
-      "layout":  WebGPU.mgrPipelineLayout.get(
+      "layout":  makePipelineLayout(
```
Comment (kainino0x) [inline src/library_webgpu.js:1196]:
```suggestion
      "layout": WebGPU.makePipelineLayout(
```

File: src/library_webgpu.js [hunk 2]
```diff
@@ -1381,7 +1386,7 @@ var LibraryWebGPU = {
 
     var desc = {
       "label": undefined,
-      "layout": WebGPU.mgrPipelineLayout.get(
+      "layout": makePipelineLayout(
```
Comment (kainino0x) [inline src/library_webgpu.js:1389]:
```suggestion
      "layout": WebGPU.makePipelineLayout(
```

**Pull request merged**